### PR TITLE
fix rally widget data fetching

### LINKED ISSUE #2725: rally collector not fetching any data
2019-02-11 13:04:09,970 INFO  com.capitalone.dashboard.Application - Started Application in 14.703 seconds (JVM running for 17.898)
2019-02-11 13:05:00,004 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:06:00,003 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:07:00,003 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:08:00,002 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:09:00,002 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:10:00,004 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:11:00,003 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:12:00,003 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally
2019-02-11 13:13:00,004 INFO  c.c.d.collector.CollectorTask - Running Collector: Rally

i dont see any error in the collector log. 

properties 

dbname=dashboarddb
dbusername=dashboarduser
dbpassword=********
dbhost=localhost
dbport=27017
rally.cron=0 0/1 * * * *
rally.servers[0]=https://rally1.rallydev.com
rally.usernames[0]=*******
rally.passwords[0]=******

 i dont see any data in feature collection in mongodb.

### LINKED ISSUE #2883: Error while fetching data from Rally
Hi, 
I am facing an issue in fetching data from Rally using the Rally collector. Below is the error message:

2019-05-09 06:20:00,635 ERROR c.c.d.collector.RallyCollectorTask - Error fetching data for the instance URL :https://rally.com
org.springframework.web.client.HttpClientErrorException: 401 SSO Redirection !

Request your help in resolving the issue.

### PR DESCRIPTION
**Pull request by yesh518**
fix #2725 fix #2883 
changed the collector type from feature to AgileTool
and also made changes in test files

### CODE REVIEW

File: api/src/main/java/com/capitalone/dashboard/request/WidgetRequest.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public Widget widget() {
             if("build".equalsIgnoreCase(name)){
                 widget.getOptions().put("buildDurationThreshold",3);
                 widget.getOptions().put("consecutiveFailureThreshold",5);
-            } else if ("feature".equalsIgnoreCase(name)) {
+            } else if ("AgileTool".equalsIgnoreCase(name)) {
```
Comment (rvema) [inline api/src/main/java/com/capitalone/dashboard/request/WidgetRequest.java]: please check for both feature and AgileTool to ensure backward compatibility.

File: api/src/test/java/com/capitalone/dashboard/request/WidgetRequestTest.java [hunk 2]
```diff
@@ -23,7 +22,7 @@ public void widgetTest() {
         allOptions.put("teamId", "321");
 
         WidgetRequest widgetRequest = new WidgetRequest();
-        widgetRequest.setName("feature");
+        widgetRequest.setName("AgileTool");
```
Comment (rvema) [inline api/src/test/java/com/capitalone/dashboard/request/WidgetRequestTest.java]: create one more test for Name as AgileTool

**Comment by rvema** [comment]
@yesh518 can you look into the compilation errors?

**Comment by yesh518** [comment]
Hi Ragha,


Good morning and i hope you having a nice day.

can you let me know exactly the compilation error that you mentioned in the request once?

________________________________

**Comment by yesh518** [comment]
i checked that error ragha and i will fix it


Thanks

Yeswanth

________________________________

**Pull request merged**